Backpack-SwiftUI/Snippet/Classes/BPKSnippet.swift

```diff
@@ -0,0 +1,104 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright © 2022 Skyscanner Ltd. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+public struct BPKSnippet: View {
+    let image: Image
+    let accessibilityLabel: String
+    let headline: String?
+    let description: String?
+    let bodyText: String?
+    let imageOrientation: ImageOrientation
+    let onClick: (() -> Void)?
+    
+    public init(
+        image: Image,
+        accessibilityLabel: String,
+        headline: String? = nil,
+        description: String? = nil,
+        bodyText: String? = nil,
+        imageOrientation: ImageOrientation = .landscape,
+        onClick: (() -> Void)? = nil
+    ) {
+        self.image = image
+        self.accessibilityLabel = accessibilityLabel
+        self.headline = headline
+        self.description = description
+        self.bodyText = bodyText
+        self.imageOrientation = imageOrientation
+        self.onClick = onClick
+    }
+    
+    public var body: some View {
+        VStack(alignment: .leading, spacing: .md) {
+            RoundedRectangle(cornerRadius: .md, style: .continuous)
+                .foregroundColor(.clear)
+                .background(
+                    image
+                        .resizable()
+                        .background(.surfaceHighlightColor)
+                        .aspectRatio(contentMode: .fill)
+                        .accessibilityLabel(accessibilityLabel)
+                )
+                .aspectRatio(aspectRatio(for: imageOrientation), contentMode: .fit)
+                .clipShape(RoundedRectangle(cornerRadius: .md, style: .continuous))
```

Comment:
Should this all be just `image` with a clipshape? here we have a RoundedRectangle that then also sets a clipshape of RoundingRectangle

Reply:
Good catch! 
Using the `image` directly (instead of being used as the "background" of this rectangle) is tricky because the image itself has a `fill` requirements, while the actual shape we want is to preserve the defined aspect ratio, needing another layer. For instance, we could use a GeometryReader like so: 

```swift
GeometryReader { geo in
    image
        .resizable()
        .scaledToFill()
        .frame(width: geo.size.width)
}
.aspectRatio(aspectRatio(for: imageOrientation), contentMode: .fill)
.clipShape(RoundedRectangle(cornerRadius: .md, style: .continuous))
```

Reply:
Lets keep it as is for now!